platform/commonUI/general/src/ui/TreeNodeView.js

```diff
@@ -1,155 +1,164 @@
+                        domainObject.getCapability('status').list()
+                    );
+                }
 
-        this.statusClasses = statuses.map(function (status) {
-            return 's-status-' + status;
-        });
+                this.labelView.model(domainObject);
+                if (this.subtreeView) {
+                    this.subtreeView.model(domainObject);
+                }
+            };
 
-        this.statusClasses.forEach(function (statusClass) {
-            this.li.addClass(statusClass);
-        }.bind(this));
-    };
-
-    TreeNodeView.prototype.model = function (domainObject) {
-        if (this.unlisten) {
-            this.unlisten();
-        }
-
-        this.activeObject = domainObject;
-
-        if (domainObject && domainObject.hasCapability('composition')) {
-            $(this.toggleView.elements()).addClass('has-children');
-        } else {
-            $(this.toggleView.elements()).removeClass('has-children');
-        }
-
-        if (domainObject && domainObject.hasCapability('status')) {
-            this.unlisten = domainObject.getCapability('status')
-                .listen(this.updateStatusClasses.bind(this));
-            this.updateStatusClasses(
-                domainObject.getCapability('status').list()
-            );
-        }
-
-        this.labelView.model(domainObject);
-        if (this.subtreeView) {
-            this.subtreeView.model(domainObject);
-        }
-    };
-
-    function getIdPath(domainObject) {
-        var context = domainObject && domainObject.getCapability('context');
-
-        function getId(domainObj) {
-            return domainObj.getId();
-        }
-
-        return context ? context.getPath().map(getId) : [];
-    }
-
-    TreeNodeView.prototype.value = function (domainObject) {
-        var activeIdPath = getIdPath(this.activeObject),
-            selectedIdPath = getIdPath(domainObject);
-
-        if (this.onSelectionPath) {
-            this.li.find('.tree-item').eq(0).removeClass('selected');
-            if (this.subtreeView) {
-                this.subtreeView.value(undefined);
-            }
-        }
-
-        this.onSelectionPath =
-            !!domainObject &&
-            !!this.activeObject &&
-            (activeIdPath.length <= selectedIdPath.length) &&
-                activeIdPath.every(function (id, index) {
-                    return selectedIdPath[index] === id;
-                });
+            function getIdPath(domainObject) {
+                var context = domainObject && domainObject.getCapability('context');
 
-        if (this.onSelectionPath) {
-            if (activeIdPath.length === selectedIdPath.length) {
-                this.li.find('.tree-item').eq(0).addClass('selected');
-            } else {
-                // Expand to reveal the selection
-                this.toggleView.value(true);
-                this.subtreeView.value(domainObject);
+                function getId(domainObj) {
```

Comment:
Should maybe be in a higher scope to make it more accessible later?

Reply:
This was already here. I have not changed that too.